packages/components/pie-checkbox/src/index.ts

```diff
@@ -147,18 +141,20 @@ export class PieCheckbox extends FormControlMixin(RtlMixin(LitElement)) implemen
             name,
             label,
             disabled,
+            disabledByParent,
```

Comment:
Just wondering why we need both `disabled` and `disabledByParent`? Would just using `disabled` be ok, or are there use cases where we need both?

Reply:
Yes, here is a slack discussion around it: https://justeat.slack.com/archives/C05DE57E27P/p1720621318557419

Summary: for the scenarios where only one checkbox in a group should be disabled we have to separate the disabled state of the checkbox from the disabled by the checkbox group.

Reply:
I guess the only thing about it being a prop, is that I see props as being an interface for consumers, but this is a prop that only really exists for `checkboxGroup` to use.

Just an idea (not sure if it would work or not), but could you handle this with events instead, so that it keeps the prop interface a bit cleaner? So when disabled is set, send an event to the checkboxes, and keep it more as a type of internal state, rather than a prop?

Reply:
@ashleynolan I rewrote the logic to use custom event dispatcher and listener